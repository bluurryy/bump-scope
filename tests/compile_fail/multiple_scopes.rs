use bump_scope::{ Bump, BumpScope };

#[allow(unused_assignments)]
fn multiple_scopes(bump: &mut Bump) {
  fn use_scope(scope: BumpScope) -> &str {
    scope.alloc_str("foo").into_ref()
  }

  let mut guard = bump.scope_guard();

  let a = use_scope(guard.scope());
  let b = use_scope(guard.scope());

  dbg!(a);
  dbg!(b);
}

fn main() {}