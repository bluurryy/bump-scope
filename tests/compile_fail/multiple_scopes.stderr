error[E0499]: cannot borrow `guard` as mutable more than once at a time
  --> tests/compile_fail/multiple_scopes.rs:12:21
   |
11 |   let a = use_scope(guard.scope());
   |                     ----- first mutable borrow occurs here
12 |   let b = use_scope(guard.scope());
   |                     ^^^^^ second mutable borrow occurs here
13 |
14 |   dbg!(a);
   |        - first borrow later used here
